src/dropdown_select.rs

```diff
@@ -0,0 +1,333 @@
+    /// Given a vector of `(label_text, enum_variant)` tuples, create a dropdown select widget
+    /// This is exactly the same interface as `Radio` so that both can be used interchangably,
+    /// with dropdown taking less space in the UI.
+    pub fn new(
+        values: impl IntoIterator<Item = (impl Into<LabelText<T>> + 'static, T)> + Clone + 'static,
+    ) -> impl Widget<T> {
+        let mut variants = Vec::new();
+        for (label, variant) in values.clone().into_iter() {
+            variants.push((label.into().display_text().to_string(), variant));
+        }
+        // A `Scope` is used here to add internal data shared within the children widgets,
+        // namely whether or not the dropdown is expanded. See `DropdownState`.
+        Scope::new(
+            DefaultScopePolicy::from_lens(
+                move |d: T| DropdownState::new(d),
+                druid::lens!(DropdownState<T>, data),
+            ),
+            Dropdown::new(
+                // TODO : Make it same width as the dropdown?
+                DropdownButton::new(move |t: &T, _: &Env| {
+                    variants
+                        .clone()
+                        .into_iter()
+                        .find_map(|(label, variant)| if *t == variant { Some(label) } else { None })
+                        .unwrap()
+                })
+                .on_click(|ctx: &mut EventCtx, t: &mut DropdownState<T>, _| {
+                    t.expanded = true;
+                    ctx.submit_notification(DROP)
+                }),
+                move |_t: &DropdownState<T>, env: &Env| {
+                    ControllerHost::new(
+                        ListSelect::new(values.clone())
+                            .lens(DropdownState::<T>::data)
+                            .border(env.get(theme::BORDER_DARK), 1.0),
+                        DropdownSelect { _t: PhantomData },
+                    )
+                },
+            ),
+        )
+    }
+}
+
+// This controller will send itself "COLLAPSE" events whenever the dropdown is removed, and
+// reacts to it by updating its expanded state
+impl<T: Data + PartialEq> Controller<DropdownState<T>, Container<DropdownState<T>>>
+    for DropdownSelect<T>
+{
+    fn event(
+        &mut self,
+        child: &mut Container<DropdownState<T>>,
+        ctx: &mut EventCtx,
+        event: &Event,
+        data: &mut DropdownState<T>,
+        env: &Env,
+    ) {
+        match event {
+            Event::Command(n) if n.is(COLLAPSE) => {
+                data.expanded = false;
+            }
+            _ => child.event(ctx, event, data, env),
+        }
+    }
+    fn lifecycle(
+        &mut self,
+        child: &mut Container<DropdownState<T>>,
+        ctx: &mut LifeCycleCtx,
+        event: &LifeCycle,
+        data: &DropdownState<T>,
+        env: &Env,
+    ) {
+        if let LifeCycle::HotChanged(false) = event {
+            ctx.submit_command(COLLAPSE);
```

Comment:
I think this will send the command to the window, which would mean multiple ones in the same window would screw each other up?

Reply:
Very good catch!

Reply:
So, you are right in principle, but actually, the dropdown is supposed to collapse as soon as the mouse exits it, so any other select dropdown in the same window would already be collapsed and the fact that they receive an additional "collapse" command would not change anything...